### DISCUSSION TITLE
Joystick usage?

### DISCUSSION THREAD

**Discussion by SteveLapp** [BroncBotz3481/YAGSL-Example#135] [Q&A]
Reviewing the example code, I am seeing that we use our x-box controller for forward, backward and side to side.   Do we need a joystick to rotate?  Or move 360?  Anyone have a decent joystick that they recommend? 

```
// CommandJoystick rotationController = new CommandJoystick(1);
  CommandJoystick driverController = new CommandJoystick(1);

  // CommandJoystick driverController   = new CommandJoystick(3);//(OperatorConstants.DRIVER_CONTROLLER_PORT);
  XboxController driverXbox = new XboxController(0);
```

**Comment by Technologyman00** [comment]
Take a look at RobotContainer.java

Lines 51 to 94 are the different included drive modes and their inputs. All use 2 joysticks. One for movement and one for rotation. You can always add your own if the drive modes don't suit you.

I've found Xbox controllers work great and the students are immediately familiar with the controls.
